src/st/no3/3C4EC.c

```diff
@@ -131,12 +131,121 @@ void EntityUnkId53(Entity* entity) {
 // large foreground tree during intro
 INCLUDE_ASM("asm/us/st/no3/nonmatchings/3C4EC", EntityForegroundTree);
 
-INCLUDE_ASM("asm/us/st/no3/nonmatchings/3C4EC", EntityUnkId50);
+void EntityUnkId50(Entity* self) {
+    Unkstruct8* roomLayout = &g_CurrentRoomTileLayout;
+    Entity* newEntity;
+    Entity* temp;
+    u16 temp_s3;
+    s32 var_v0;
+    u16* ptr;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_80180AD0);
+        ptr = D_801814EC;
+        temp = &g_Entities[192];
+        self->unk68 = 0xC0;
+        self->ext.generic.unk7C.s = 0;
```

Comment:
Should we make an extension?

Reply:
For just one member? I think it's better to do that at documentation time because that can be a member used by other function related to this one and we don't have those decompiled yet.